pkg/server/api_registration.go

```diff
@@ -83,15 +83,24 @@ func (i *InventoryServer) apiRegistration(resp http.ResponseWriter, req *http.Re
 	inventory, err := i.authMachine(conn, req, registration.Namespace)
 	if err != nil {
 		log.Errorf("authentication failed: ", err)
+		http.Error(resp, "authentication failure", http.StatusUnauthorized)
 		return err
 	}
 	// no error and no inventory: Auth header is missing or unrecognized
 	if inventory == nil {
-		log.Warningf("websocket connection: unrecognized authentication method")
+		if authHeader := req.Header.Get("Authorization"); authHeader != "" {
+			errMsg := "unrecognized authentication method"
+			log.Errorf("websocket connection: %s", errMsg)
+			http.Error(resp, errMsg, http.StatusUnauthorized)
+			return err
+		}
+
+		log.Warning("unauthenticated websocket connection")
 		if err = i.unauthenticatedResponse(registration, resp); err != nil {
```

Comment:
Saw the unauthenticatedResponse is also called in line 67, which seams weird to me as errors out, but returns the registration yaml.

In any case this are details we can improve later, it is not really related with this PR.

Reply:
No, I think there it is fine: the error at line 67 is because we are promoting the connection to a websocket from the server side, but the client didn't added the "Connection: Upgrade" header, which is mandatory for promoting an HTTP GET to a websocket connection. That tells us that it really was a plain, unauthenticated HTTP GET, which should get back the registration yaml.
Here instead I log a Warning because we don't expect a websocket connection to get the registration yaml. We could just return error here, but I see no harm to return the registration yaml also on websocket connections (also if I don't see why one should download the registration yaml via a websocket connection honestly).

Reply:
> unauthenticated HTTP GET, which should get back the registration yaml.

This is the behavior that feels weird to me. Having a different valid behavior for (un)authenticated HTTP GET calls over the same URL. Anyway, unrelated to this PR 😉